File: scala-play-server/app/model/BasicStream.scala
package model

import play.api.libs.json._

/**
  * Represents the Swagger definition for BasicStream.
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2021-08-09T21:37:15.417726Z[Etc/UTC]")
case class BasicStream(
  streamId: Option[OasAnyTypeNotMapped],
  name: Option[OasAnyTypeNotMapped],
  description: Option[OasAnyTypeNotMapped],
  dateCreated: Option[OasAnyTypeNotMapped],
  inviteOnly: Option[OasAnyTypeNotMapped],
  renderedDescription: Option[OasAnyTypeNotMapped],
  isWebPublic: Option[OasAnyTypeNotMapped],
  streamPostPolicy: Option[OasAnyTypeNotMapped],
  messageRetentionDays: Option[OasAnyTypeNotMapped],
  historyPublicToSubscribers: Option[OasAnyTypeNotMapped],
  firstMessageId: Option[OasAnyTypeNotMapped],
  isAnnouncementOnly: Option[OasAnyTypeNotMapped]
)

object BasicStream {
  implicit lazy val basicStreamJsonFormat: Format[BasicStream] = Json.format[BasicStream]
}

File: clojure/src/zulip_rest_api/specs/basic_stream.clj
(ns zulip-rest-api.specs.basic-stream
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [zulip-rest-api.specs. :refer :all]
            [zulip-rest-api.specs. :refer :all]
            [zulip-rest-api.specs. :refer :all]
            [zulip-rest-api.specs. :refer :all]
            [zulip-rest-api.specs. :refer :all]
            [zulip-rest-api.specs. :refer :all]
            [zulip-rest-api.specs. :refer :all]
            [zulip-rest-api.specs. :refer :all]
            [zulip-rest-api.specs. :refer :all]
            [zulip-rest-api.specs. :refer :all]
            [zulip-rest-api.specs. :refer :all]
            [zulip-rest-api.specs. :refer :all]
            )
  (:import (java.io File)))


(def basic-stream-data
  {
   (ds/opt :stream_id) any-type-spec
   (ds/opt :name) any-type-spec
   (ds/opt :description) any-type-spec
   (ds/opt :date_created) any-type-spec
   (ds/opt :invite_only) any-type-spec
   (ds/opt :rendered_description) any-type-spec
   (ds/opt :is_web_public) any-type-spec
   (ds/opt :stream_post_policy) any-type-spec
   (ds/opt :message_retention_days) any-type-spec
   (ds/opt :history_public_to_subscribers) any-type-spec
   (ds/opt :first_message_id) any-type-spec
   (ds/opt :is_announcement_only) any-type-spec
   })

(def basic-stream-spec
  (ds/spec
    {:name ::basic-stream
     :spec basic-stream-data}))
